activerecord/test/cases/relations_test.rb

```diff
@@ -1909,6 +1909,12 @@ def test_presence
     end
   end
 
+  test "unloaded relations add a LIMIT in #inspect but not in #to_a" do
+    relation = Post.all
+    assert_sql(/LIMIT/) { relation.inspect }
+    assert_sql(/^(?!.*LIMIT)/) { relation.to_a }
```

Comment:
Wondering, would it be better to sacrifice another bit of readability for correctness by using `\A(?!.*LIMIT)/m`, or am I over-thinking it?